tests/dns.rs

```diff
@@ -0,0 +1,114 @@
+use std::{
+    io::{BufRead, BufReader, Write},
+    net::TcpStream,
+    process::{Child, Command},
+    sync::{
+        mpsc,
+        mpsc::{Receiver, Sender},
+    },
+    thread,
+    time::Duration,
+};
+
+fn start_server() -> (Child, Receiver<String>) {
+    let (log_sender, log_receiver): (Sender<String>, Receiver<String>) = mpsc::channel();
+    let mut directoryd_process = Command::new("./target/debug/directoryd")
+        .stdout(std::process::Stdio::piped())
+        .spawn()
+        .unwrap();
+
+    let stdout = directoryd_process.stdout.take().unwrap();
+    let sender = log_sender.clone();
+    thread::spawn(move || {
+        let reader = BufReader::new(stdout);
+        reader.lines().map_while(Result::ok).for_each(|line| {
+            sender.send(line).unwrap_or_else(|e| {
+                println!("Failed to send log: {}", e);
+            });
+        });
+    });
+
+    thread::sleep(Duration::from_secs(5));
+    (directoryd_process, log_receiver)
+}
+
+fn wait_for_server_start(log_receiver: &Receiver<String>) {
+    let mut server_started = false;
+    while let Ok(log_message) = log_receiver.recv_timeout(Duration::from_secs(5)) {
+        if log_message.contains("RPC socket binding successful") {
+            server_started = true;
+            break;
+        }
+    }
+    assert!(
+        server_started,
+        "Server did not start within the expected time"
+    );
+    thread::sleep(Duration::from_secs(10));
```

Comment:
Why another 10 second pause? We are already waiting on the loop above for server start.